tests/integration_tests/without_db/udfs/test_model_downloader_udf.py

```diff
@@ -76,49 +69,38 @@ def __init__(self, id: str, url_localfs: str, token_conn_name: str):
         }
         self.model_path = bucketfs_operations.get_bucketfs_model_save_path(
             self.sub_dir, self.tiny_model)
-        self.bucketfs_connection = Connection(
-            address=f"{url_localfs}/bucket{id}",
-            user=None,
-            password=None
-        )
+        self.bucketfs_connection = create_mounted_bucketfs_connection(tmp_dir, f'bucket{id}')
         self.token_connection = None if not self.token_conn_name \
-            else Connection(address=f"", password="valid")
-
-    @property
-    def bucketfs_location(self):
-        return BucketFSFactory().create_bucketfs_location(
-            url=self.bucketfs_connection.address,
-            user=self.bucketfs_connection.user,
-            pwd=self.bucketfs_connection.password)
+            else create_hf_token_connection("valid")
 
     def list_files_in_bucketfs(self):
-        return self.bucketfs_location.list_files_in_bucketfs(
-            str(self.sub_dir))
-
-
-def test_model_downloader_udf_implementation():
-    with tempfile.TemporaryDirectory() as tmpdir_name:
-        url_localfs = f"file://{tmpdir_name}/bucket"
-        env1 = TestEnvironmentSetup(
-            "1", url_localfs, token_conn_name='')
-        env2 = TestEnvironmentSetup(
-            "2", url_localfs, token_conn_name='token_conn_name')
-
-        ctx = Context([env1.ctx_data, env2.ctx_data])
-        exa = ExaEnvironment({
-            env1.bucketfs_conn_name: env1.bucketfs_connection,
-            env2.bucketfs_conn_name: env2.bucketfs_connection,
-            env2.token_conn_name: env2.token_connection
-        })
-
-        # run udf implementation
-        model_downloader = ModelDownloaderUDF(exa)
-        model_downloader.run(ctx)
-
-        # assertions
-        env1_bucketfs_files = env1.list_files_in_bucketfs()
-        env2_bucketfs_files = env2.list_files_in_bucketfs()
-        assert ctx.get_emitted()[0] == (str(env1.model_path), str(env1.model_path.with_suffix(".tar.gz")))
-        assert ctx.get_emitted()[1] == (str(env2.model_path), str(env2.model_path.with_suffix(".tar.gz")))
-        assert str(Path(ctx.get_emitted()[0][1]).relative_to(env1.sub_dir)) in env1_bucketfs_files
-        assert str(Path(ctx.get_emitted()[1][1]).relative_to(env2.sub_dir)) in env2_bucketfs_files
+        bucketfs_location = bucketfs_operations.create_bucketfs_location_from_conn_object(
+            self.bucketfs_connection)
+        return [str(node[0] / leaf) for node in bucketfs_location.walk()
+                for leaf in node[2] if not node[1]]
```

Comment:
having named constants for the indexes `0,1,2` would improve readability significantly here.

Reply:
Made a change as per the suggestion and moved this code to a separate function. The function is also used in another test.